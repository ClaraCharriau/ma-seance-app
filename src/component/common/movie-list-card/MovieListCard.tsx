import { NavLink } from 'react-router-dom';
import { Movie } from '../../../models/Movie';
import style from './MovieListCard.module.css';
import useDurationFormat from '../../../hooks/useDurationFormat';
import useDateDMYFormat from '../../../hooks/dates/useDateDMYFormat';
import useStringListFormat from '../../../hooks/useStringListFormat';
import config from '../../../config/config.helper';

interface MovieListCardProps {
    movie: Movie;
}

const MovieListCard = (props: MovieListCardProps) => {
    const { movie } = props;
    const TMDB_PATH = config.tmdbImgPath.medium;
    const formattedDuration = useDurationFormat(movie.duration);
    const formattedReleaseDate = useDateDMYFormat(movie.releaseDate);
    const directors = useStringListFormat(movie.directors);
    const cast = useStringListFormat(movie.cast);

    return (
        <NavLink to={`/movies/${movie.id}`} state={{ id: movie.id }} className={style.movieCard}>
            <img
                src={TMDB_PATH + movie.posterLink}
                alt={`affiche du film ${movie.title}`}
                className={style.movieCardImg}
            />
            <div>
                <p className={style.movieTitle}>{movie.title}</p>
                <p className={style.movieDateDuration}>
                    {formattedReleaseDate} - {formattedDuration}
                </p>
                <p className={style.movieDirector}>Réalisé par {directors}</p>
                <p className={style.movieCast}>Avec {cast}</p>
            </div>
        </NavLink>
    );
};

export default MovieListCard;
