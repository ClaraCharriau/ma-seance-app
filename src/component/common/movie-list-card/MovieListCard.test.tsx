import mockMovie from '../../../mocks/movies/movie-1.json';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import MovieListCard from './MovieListCard';

describe('Movie list card component test', () => {
    it('should render component a card', () => {
        // Given
        // When
        const component = render(
            <BrowserRouter>
                <MovieListCard movie={mockMovie} />
            </BrowserRouter>
        );

        // Then
        expect(component.baseElement).toMatchSnapshot();
    });
});
