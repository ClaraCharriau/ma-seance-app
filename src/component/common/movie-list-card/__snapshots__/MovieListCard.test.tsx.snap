// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie list card component test should render component a card 1`] = `
<body>
  <div>
    <a
      class="movieCard"
      href="/movies/3d8f1342-15f1-44b1-a48f-4581d654b94a/day-1"
    >
      <img
        alt="affiche du film Pauvres créatures"
        class="movieCardImg"
        src="https://image.tmdb.org/t/p/w500/5fT98da9ccWN2xr8VOJrSBp3Cdw.jpg"
      />
      <div>
        <p
          class="movieTitle"
        >
          Pauvres créatures
        </p>
        <p
          class="movieDateDuration"
        >
          17 janvier 2024
           - 
          1h30
        </p>
        <p
          class="movieDirector"
        >
          Réalisé par 
          Yorgos Lanthimos
        </p>
        <p
          class="movieCast"
        >
          Avec 
          Emma Stone, Mark Ruffalo, Willem Dafoe
        </p>
      </div>
    </a>
  </div>
</body>
`;
